### DISCUSSION TITLE
変数の値渡しについて

### DISCUSSION THREAD

**Discussion by silli-3072** [lit-iphone-course/lit-iphone-course#23] [Question-original] [answered]
お疲れ様です。ToDoの詳細画面の実装について質問です。
テーブルビューの値を取得し、その行を変数で宣言→遷移先の画面に変数を持ち出し対応する部分のレルムデータの持ち出し&表示
という流れで実装しようと考えました。
現状できていること
✅遷移前の画面で選択行を変数に入れる
✅遷移先の画面で詳細を表示する
できなかったこと
✅テーブルビューに設置したボタンを押した動作での選択行の取得→ボタンのメソッドを下記のように実装しましたがエラーが出てしまいます。ボタン動作での行取得方法を知りたいです。
```
@IBAction func cellButton() {
        print("🎀",cellNumber)
        
        cellNumber = indexPath.row
        
    }
```
✅変数の値渡し→[参考記事](https://qiita.com/hasumi024/items/8f5f52c9a634b3c360aa)をもとに実装してみましたが、遷移先の画面で`var cellNumber = 0`と宣言しており、持ち出された値ではなく０が採用されており変数の初期値を設定しない方法を知りたいです。

こちら[GitHub](https://github.com/silli-3072/LA4_ToDoApp)になります。回答お願いします。

**Comment by txxxxc** [comment, marked as answer]
> テーブルビューに設置したボタンを押した動作での選択行の取得→ボタンのメソッドを下記のように実装しましたがエラーが出てしまいます。ボタン動作での行取得方法を知りたいです。

`performSegue(withIdentifier:sender:)`っていうメソッドがあるので、それを使って`didSelectRowAt`の中で画面遷移してあげればcellの番号を渡しながら画面遷移できるのでやってみてー！

参考↓
[【Swift】画面遷移の方法まとめ - Qiita](https://qiita.com/misakiagata/items/b7f6c2f6c9f988ec38c7)

**Reply by silli-3072** [reply]
初め、いただいた記事と同じものを見て
```
func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        cellNumber = indexPath.row
        
        self.performSegue(withIdentifier: "toDetailViewController", sender: nil)
        
        print("🎀",cellNumber)
    }
```
このように実装していました。ただ、ビルドするとエラーになってしまい解決できず今のようなボタンを使っての遷移をすることにしました。上記のコードでエラーが出る要因はなんでしょうか？
上記のコードの状態で再度[コミット](https://github.com/silli-3072/LA4_ToDoApp)しました。確認お願いします。

**Reply by txxxxc** [reply]
なるほど！
どんなエラー出たかスクショかコピペで教えてほしいです！！

**Reply by silli-3072** [reply]
Thread 1: "Receiver (<ToDoApp.ViewController: 0x15dd0a6c0>) has no segue with identifier 'toDetailViewController'"

これです！

**Reply by txxxxc** [reply]
あ、それ普通にidをsegueに設定していないだけやからid振って試してみてー！！！

**Reply by silli-3072** [reply]
！！！
できました〜〜
